node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+            pending_payable_opt: None,
+        };
+        let balance_2 = multiple_by_billion(2_000_000);
+        let account_2 = PayableAccount {
+            wallet: make_wallet("def"),
+            balance_wei: balance_2,
+            last_paid_timestamp: now.checked_sub(Duration::from_secs(50_000)).unwrap(),
+            pending_payable_opt: None,
+        };
+        let balance_3 = multiple_by_billion(5_000_000);
+        let account_3 = PayableAccount {
+            wallet: make_wallet("ghi"),
+            balance_wei: balance_3,
+            last_paid_timestamp: now.checked_sub(Duration::from_secs(70_000)).unwrap(),
+            pending_payable_opt: None,
+        };
+        let payables = vec![account_1, account_2, account_3];
+        let qualified_payables =
+            make_guaranteed_qualified_payables(payables, &PRESERVED_TEST_PAYMENT_THRESHOLDS, now);
+        let calculator_mock = CriterionCalculatorMock::default()
+            .calculate_result(multiple_by_billion(2_000_000_000))
+            .calculate_result(0)
+            .calculate_result(0);
+        let mut subject = PaymentAdjusterReal::new();
+        subject.calculators.push(Box::new(calculator_mock));
+        subject.logger = Logger::new(test_name);
+        let agent_id_stamp = ArbitraryIdStamp::new();
+        let cw_service_fee_balance_minor = balance_2;
+        let disqualification_arbiter = &subject.disqualification_arbiter;
+        let agent_for_analysis = BlockchainAgentMock::default()
+            .agreed_transaction_fee_margin_result(*TRANSACTION_FEE_MARGIN)
+            .service_fee_balance_minor_result(cw_service_fee_balance_minor)
+            .transaction_fee_balance_minor_result(U256::MAX)
+            .estimated_transaction_fee_per_transaction_minor_result(12356);
+        let analysis_result = subject.analyzer.analyze_accounts(
+            &agent_for_analysis,
+            disqualification_arbiter,
+            qualified_payables,
+            &subject.logger,
+        );
+        // If the initial analysis at the entry into the PaymentAdjuster concludes there is no point
+        // going off because even the least demanding account could not be satisfied, and we would
+        // get an error here.
+        // However, it can only assess the lowest disqualification limit of an account in that set.
+        // Probably not as usual, but this particular account can be later outplayed by another one
+        // that is equipped with some extra significance while its disqualification limit does not
+        // fit inder the consuming wallet balance anymore. A late error, possibly two different, is
+        // born.
+        let adjustment_analysis = match analysis_result {
+            Ok(Either::Right(analysis)) => analysis,
+            x => panic!(
+                "We expected to be let it for an adjustments with AnalyzedAccounts but got: {:?}",
+                x
+            ),
+        };
+        let agent = {
```

Comment:
At the moment, it seems doable to use the old agent. Also, we're not even asserting on the id stamp, since all we want to look at is the error.